Sources/Sentry/SentryHttpTransport.m

```diff
@@ -84,36 +92,47 @@ - (void)sendEnvelope:(SentryEnvelope *)envelope
 // flow
 - (void)sendAllCachedEnvelopes
 {
-    if (![self.requestManager isReady]) {
-        return;
+    @synchronized(self) {
+        if (self.isSending || ![self.requestManager isReady]) {
+            return;
+        }
+
+        self.isSending = YES;
     }
 
-    for (SentryFileContents *fileContents in [self.fileManager getAllEnvelopes]) {
-        SentryEnvelope *envelope = [SentrySerialization envelopeWithData:fileContents.contents];
-        if (nil == envelope) {
-            [self.fileManager removeFileAtPath:fileContents.path];
-            continue;
+    SentryFileContents *oldestEnvelope = [self.fileManager getOldestEnvelope];
+    if (nil == oldestEnvelope) {
+        @synchronized(self) {
+            self.isSending = NO;
         }
+        return;
+    }
 
-        SentryEnvelope *rateLimitedEnvelope =
-            [self.envelopeRateLimit removeRateLimitedItems:envelope];
-        if (rateLimitedEnvelope.items.count == 0) {
-            [self.fileManager removeFileAtPath:fileContents.path];
-            continue;
-        }
+    SentryEnvelope *envelope = [SentrySerialization envelopeWithData:oldestEnvelope.contents];
+    if (nil == envelope) {
+        [self.fileManager removeFileAtPath:oldestEnvelope.path];
+        [self sendNextCachedEnvelope];
+        return;
+    }
 
-        NSError *requestError = nil;
-        NSURLRequest *request = [self createEnvelopeRequest:rateLimitedEnvelope
-                                           didFailWithError:requestError];
-
-        if (nil != requestError) {
-            [SentryLog logWithMessage:requestError.localizedDescription
-                             andLevel:kSentryLogLevelError];
-            [self.fileManager removeFileAtPath:fileContents.path];
-            continue;
-        } else {
-            [self sendCached:request withFilePath:fileContents.path];
-        }
+    SentryEnvelope *rateLimitedEnvelope = [self.envelopeRateLimit removeRateLimitedItems:envelope];
+    if (rateLimitedEnvelope.items.count == 0) {
+        [self.fileManager removeFileAtPath:oldestEnvelope.path];
+        [self sendNextCachedEnvelope];
+        return;
+    }
+
+    NSError *requestError = nil;
+    NSURLRequest *request = [self createEnvelopeRequest:rateLimitedEnvelope
+                                       didFailWithError:requestError];
+
+    if (nil != requestError) {
+        [SentryLog logWithMessage:requestError.localizedDescription andLevel:kSentryLogLevelError];
+        [self.fileManager removeFileAtPath:oldestEnvelope.path];
+        [self sendNextCachedEnvelope];
+        return;
+    } else {
+        [self sendCached:request withFilePath:oldestEnvelope.path];
```

Comment:
This seem to be the only use of `sendCache`, is that right? if so, can we inline to avoid having more private methods accessing the flag under a lock?